Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -55,6 +57,14 @@ public float FlatHandThreshold
             set { flatHandThreshold = value; }
         }
 
+        [SerializeField]
+        [Tooltip("With this active, solver will follow hand rotation until the menu is sufficiently aligned with the gaze, at which point it faces the camera.")]
+        private bool followHandUntilFacingCamera = false;
+
+        [SerializeField]
+        [Tooltip("Angle (in degrees) between hand up and camera forward, below which the hand menu follows the gaze, if followHandUntilFacingCamera is active.")]
+        private float followHandCameraFacingThresholdAngle = 60f;
```

Comment:
It would be [good to write a test](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#prioritize-writing-tests) to ensure this behavior doesn't regress from somebody else's future changes. [Here is some information for how to write an run tests](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/UnitTests.html).

Reply:
Hmm... I don't really know how to write a test that might make sense in this case.
Do you think writing a test is a blocking part of this pr?

Reply:
I think a basic test would be test starting below the threshold, crossing it, and then crossing back below the threshold. And between these changes validate that the solver positions the object at the expected different places (in this case hand vs camera following)

Reply:
If someone writes a test for the hand constraint class, I'll happily add a function to test my changes. I just don't have the time to write up an entire test class for it right now.